core/src/utils/env.rs

```diff
@@ -0,0 +1,86 @@
+use aligned_sdk::core::types::Chain;
+extern crate dotenv;
+use dotenv::dotenv;
+use log::debug;
+
+use super::constants::{
+    ANVIL_BATCHER_ADDR, ANVIL_BATCHER_ETH_ADDR, ANVIL_ETH_RPC_URL, ANVIL_PROOF_GENERATOR_ADDR,
+};
+
+pub struct EnvironmentVariables {
+    pub rpc_url: String,
+    pub chain: Chain,
+    pub batcher_addr: String,
+    pub batcher_eth_addr: String,
+    pub eth_rpc_url: String,
+    pub proof_generator_addr: String,
+    pub keystore_path: Option<String>,
+    pub private_key: Option<String>,
+}
+
+impl EnvironmentVariables {
+    pub fn new() -> Result<EnvironmentVariables, String> {
+        dotenv().map_err(|err| format!("Couldn't load .env file: {}", err))?;
+
+        let rpc_url = std::env::var("MINA_RPC_URL")
+            .map_err(|err| format!("Couldn't get MINA_RPC_URL env. variable: {err}"))?;
+        let chain = match std::env::var("ETH_CHAIN")
+            .map_err(|err| format!("Couldn't get ETH_CHAIN env. variable: {err}"))?
+            .as_str()
+        {
+            "devnet" => {
+                debug!("Selected Anvil devnet chain.");
+                Chain::Devnet
+            }
+            "holesky" => {
+                debug!("Selected Holesky chain.");
+                Chain::Holesky
+            }
+            _ => return Err(
+                "Unrecognized chain, possible values for ETH_CHAIN are \"devnet\" and \"holesky\"."
+                    .to_owned(),
+            ),
+        };
+
+        // Default value is only valid for Anvil devnet setup.
+        let load_var_or = |key: &str, default: &str| -> Result<String, String> {
+            match std::env::var(key) {
+                Ok(value) => Ok(value),
+                Err(_) if matches!(chain, Chain::Devnet) => {
+                    debug!("Using default {} for devnet: {}", key, default);
+                    Ok(default.to_string())
+                }
+                Err(err) => Err(format!(
+                    "Chain selected is not Devnet but couldn't read {}: {}",
+                    key, err
+                )),
+            }
+        };
```

Comment:
Because this closure doesn't use vars from the function where it's defined, I think it's better to declare a private function instead of a closure.

Reply:
It actually uses the `chain` variable but I agree that a closure is unnecessary in this case, will refactor it!